### DISCUSSION TITLE
Controller action is not working with 404 error

### DISCUSSION THREAD

**Discussion by kudryavtsevda** [OrchardCMS/OrchardCore#11364] [Q&A] [answered]
I am trying to render cshtml view via standart approach: controllers and razor view. But it is not working with 404 error.
Module name: OrchardCoreContrib.Html, Controller name: GrapesJSController, View name: Index
This is startup.cs
![image](https://user-images.githubusercontent.com/4447809/157930217-308ee2f0-70ca-42b1-acc3-77bc5dba7911.png)
This is GrapesJSController
![image](https://user-images.githubusercontent.com/4447809/157930560-b07cd1af-ed1d-4747-8a45-499536bb3b92.png)
This is the simplest view:
![image](https://user-images.githubusercontent.com/4447809/157930662-997c5839-94f7-4ecf-bb33-eb557168f756.png)
This is the manifest
![image](https://user-images.githubusercontent.com/4447809/157930748-820e5cde-3831-4f1c-97a6-0d3ec312fd06.png)

And this is the bad result:
![image](https://user-images.githubusercontent.com/4447809/157930962-db7c67ce-4949-4bff-8a50-224131f05dce.png)

Please tell me anybody what am I doing wrong?

**Comment by Skrypt** [comment]
Have you added the OrchardCore.Contrib.Html module to your Mysite.Web project as a reference?
Have you enabled the module feature in the admin?

**Reply by kudryavtsevda** [reply]
> Have you added the OrchardCore.Contrib.Html module to your Mysite.Web project as a reference? Have you enabled the module feature in the admin?

Yes, I have added it and enabled. Moreover this module works good apart from this issue. 
![image](https://user-images.githubusercontent.com/4447809/157935605-6869d448-7578-4d4a-94cf-b213f7b09a76.png)

Also I noticed that due to some reason startup.cs of the module is not invoked. I have no idea why. But as a result this route is not in the list of routes. 
![image](https://user-images.githubusercontent.com/4447809/157935704-a1076cfd-2fa4-448b-8597-31f1ba40999f.png)

**Comment by hishamco** [comment]
Can you change the route name and have a try?

**Reply by kudryavtsevda** [reply]
> Can you change the route name and have a try?

I spent all day. Belive I tried everything. BUT I found the root of the problem: wrong manifest here:
![image](https://user-images.githubusercontent.com/4447809/157948588-4a29c6c8-07da-481a-9a91-577436b60f7c.png)

Feature name should be ` OrchardCoreContrib.Html` as per this explanation:
![image](https://user-images.githubusercontent.com/4447809/157948988-0b63f5c7-e723-47fb-9806-f9b4d76bf246.png)

**Reply by hishamco** [reply]
I'm suggesting to transfer the issue to OCC repo

**Reply by kudryavtsevda** [reply]
> I'm suggesting to transfer the issue to OCC repo

I didn't get well. Could you explain?

**Reply by hishamco** [reply]
What I mean let us create another issue in OCC repository, while the issue related to OCC not OC

**Reply by hishamco** [reply]
> I spent all day. Belive I tried everything. BUT I found the root of the problem: wrong manifest here:

I have similar one here with a controller

https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/b039a3daee2a571c237ffebf64e7fc8cb2a2cc6d/src/OrchardCoreContrib.Users/Manifest.cs#L10-L15

**Reply by Skrypt** [reply]
The Manifest.cs file has an error like @hishamco is mentioning. You should name your Module the same as your assembly name. So in that case OrchardCoreContrib.Html

**Reply by kudryavtsevda** [reply]
> The Manifest.cs file has an error like @hishamco is mentioning. You should name your Module the same as your assembly name. So in that case OrchardCoreContrib.Html

thank you for your assistance but let me correct you: it is me who found the root of the issue on my last screenshot

**Reply by hishamco** [reply]
Seems the issue because we have a single feature, @kudryavtsevda let me double check again

**Reply by kudryavtsevda** [reply]
> Seems the issue because we have a single feature, @kudryavtsevda let me double check again

Yes, the author of tutorial claims: The technical ID of the feature. The base feature should have the same name as the module (i.e. its csproj file).
But for the second feature he says: We commonly suffix the base feature/module ID and name for sub-features.

So it is explicitly outlined and I would not consider it is as I issue or bug. Just only as an inconvenience which makes me throw all day long in a trash

**Reply by Skrypt** [reply]
You don't need to have a module and feature for this simple module. This is what we meant. You will generally use a feature inside a module if you have sub-features that can be disabled. Just, for example, the Lucene module has a Search feature which is not required for the Lucene module to work.

**Comment by Skrypt** [comment, marked as answer]
The Manifest.cs file has an error like @kudryavtsevda is mentioning. You should name your Module the same as your assembly name. So in that case OrchardCoreContrib.Html

**Comment by jtkech** [comment]
Yes, if a module has only one feature it is not required to use the Feature attribute, this only one main feature is created automatically with the module Id, but it's okay to still use the Feature attribute so that it is easier to add other Features afterwards, but then, yes it is strongly recommended that the first Feature (or only one) matches the module Id. Anyway, here at least the route area should match the related feature Id.

**Reply by jtkech** [reply]
yes, @kudryavtsevda found the issue !!!

**Comment by hishamco** [comment]
Seems it's my bad to change the features IDs serveral days ago, I remember that change done when I planned to add extra feature but I didn't

@kudryavtsevda I will publish a patch release soon, also you can check that the Id has been changed serveral days ;)

Thanks again

**Comment by hishamco** [comment]
Wait a minute @kudryavtsevda seems I didn't publish the package yet, it's WIP PR https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/pull/36, but again I need to fix the OCC.Users modules, don't blame me because it's WIP ;)

**Comment by hishamco** [comment]
@jtkech could you please check my comment [here](https://github.com/OrchardCMS/OrchardCore/discussions/11364#discussioncomment-2341851), I just checked the source, everything seems to work fine

**Reply by jtkech** [reply]
@hishamco 

Yes in place of using `OCC.Users.Impersonation` for your first and only one feature, and then having to decorate your startup and controller with `[Feature("OCC.Users.Impersonation")]`, better to just use `OCC.Users` without having to decorate anything. But then in any case, your route area should match the related feature Id that you will use at the end.

**Reply by hishamco** [reply]
That means nothing wrong with what I did, coz I'm planned to add extra features, which requires me to such decoration for every feature, right?

**Reply by jtkech** [reply]
Yes but usually we don't do that for the first feature

And currently your route area `OCC.Users` doesn't match your feature Id `OCC.Users.Impersonation`

**Comment by masterflylox** [comment]
Does the orchard sovereign archive still exist? Where can I view it? On github? What is the link? I get a 404 error every link.

**Reply by Piedone** [reply]
What are you referring to by "orchard sovereign archive"?
